packages/coinstac-ui/app/render/log.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const loadConfig = require('../config.js');
+const os = require('os');
+const path = require('path');
+const readLastLines = require('read-last-lines');
+const { Tail } = require('tail');
+
+function maybeAddOutput(output, className) {
+  if (typeof document !== 'undefined') {
+    const outputEl = document.getElementById('output');
+    const code = document.createElement('code');
+    code.className = className;
+    code.innerHTML = output;
+    outputEl.appendChild(code);
+    // http://stackoverflow.com/a/270628
+    outputEl.scrollTop = outputEl.scrollHeight;
+  }
+}
+
+function logData(data) {
+  console.log(data); // eslint-disable-line no-console
+  maybeAddOutput(data);
+}
+
+function logError(error) {
+  console.error(error); // eslint-disable-line no-console
+  maybeAddOutput(error.message, 'error');
+}
+
+loadConfig()
+  .then((config) => {
+    const target = path.join(
+      os.homedir(),
+      config.get('logLocations')[os.platform()],
+      config.get('logFile')
+    );
+
+    console.log(`Reading log: ${target}`); // eslint-disable-line no-console
+
+    return Promise.all([target, readLastLines.read(target, 50)]);
```

Comment:
[tail](https://www.npmjs.com/package/tail) doesn’t support a `-n`-style number flag: it either reads the entire log file – :scream: – or watches for new lines. [read-last-lines](https://www.npmjs.com/package/read-last-lines) dumps the last 50 lines here.